portal/server/routes/user/profile.js

```diff
@@ -46,18 +46,57 @@ router.get('/:_id/change-password', async (req, res) => {
 router.post('/:_id/change-password', async (req, res) => {
   // Ensure that the user is logged in
   if (req?.session?.user) {
+    let formattedValidationErrors;
+
     const { _id } = requestParams(req);
+    const { currentPassword, password, passwordConfirm } = req.body;
 
-    const { status, data } = await api.updateUser(_id, req.body, req.session.userToken);
+    if (!currentPassword || !password || !passwordConfirm) {
```

Comment:
do we have a test for this?

Reply:
Have added extensive tests covering both user change their own password and password reset (upon forgot password email).